### ISSUE TITLE
NethVoice restore config fails in new systems

### ISSUE THREAD

**Issue by DavidePrincipi** [nethesis/dev#5885]
Restoring a configuration backup fails due to early start of `mysqld.service`.

**Steps to reproduce**

- Install a clean NethServer Enterprise system
- Register
- Update
- Restore a configuration backup with NethVoice 14. Ensure "Restore network" is set.

**Expected behavior**

The restore is successful

**Actual behavior**

The restore-config fails. 

![image](https://user-images.githubusercontent.com/2920838/95605098-c1d08480-0a58-11eb-92e7-8b0a8481bd2e.png)

1. From the log trace, it seems `nethcti-server` is started within `interface-update`, by

        /etc/e-smith/events/trusted-networks-modify/services2adjust/nethcti-server

2. The `nethcti-server` unit triggers the start of `asterisk` and `mysqld` as chained dependencies, before `nethserver-mysql-update` runs. See https://gist.github.com/DavidePrincipi/c41cdf5e8455750ccffdbf20a0c9bf38#file-messages-L262-L286

3. As /var/lib/mysql is now initialized, the nethserver-mysql-conf action exits without setting the root password as expected. See https://gist.github.com/DavidePrincipi/c41cdf5e8455750ccffdbf20a0c9bf38#file-messages-L1079

There are also further secondary failures, all caused by the wrong `mysql` DB root password.

**Components**

- nethserver-mysql-1.1.4-1.ns7.noarch
- nethserver-phonebook-mysql-2.2.3-1.ns7.noarch
- nethvoice-module-nethvplan14-14.2.2-1.ns7.noarch
- nethcti-server3-3.20.2-1.ns7.x86_64
- nethserver-nethvoice14-14.7.1-1.ns7.noarch

**See also**

If nethserver-mysql was previously installed (e.g. NethVoice box image) this issue fixes the root password properly:

https://github.com/nethserver/dev/issues/5339

**Comment by DavidePrincipi** [comment]
**Test case 0**

[see below...]

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethserver-nethvoice14-14.7.1-1.4.gd251f40.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethserver-nethvoice14-14.7.1-1.4.gd251f40.ns7.noarch.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethcti-server3-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm) 
- nethcti-server3-debuginfo-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-debuginfo-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethserver-nethvoice14-14.7.1-1.6.g87f0ac7.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethserver-nethvoice14-14.7.1-1.6.g87f0ac7.ns7.noarch.rpm)

**Comment by DavidePrincipi** [comment]
**Test case 1**

Check the bug is not reproducible. To override the stable packages, 


* install and configure additional YUM repositories with the testing packages. You can run the following command, to enable testing repositories

       mkdir -p /etc/e-smith/templates-custom/etc/nethserver/eorepo.conf/ && \
       echo -n '{ push @repos, "nethesis-testing", "nethserver-testing"; $OUT=""; }' > /etc/e-smith/templates-custom/etc/nethserver/eorepo.conf/40testing && \
       signal-event software-repos-save

 *  or download the testing packages in the current directory and run the [update-local-repo](https://gist.github.com/DavidePrincipi/ef196d05287c46be716cf9e20bf66fd0) command to cherry-pick the RPMs. 

Additional checks after the restore-config works:
- Check that the action `nethserver-freepbx-backup-restore` runs without any "no such file or directory" error.
- Stop `asterisk.service` and check that `nethvoice-alters.service` does not start it again after 10 minutes.


**Test case 2**

Check against regressions in affected use cases:

1. Change the trusted networks in a running NethVoice system. `nethcti-server` must be reloaded.
2. Change the first green IP address in a running NethVoice system: FreePBX/Asterisk must receive the new IP (unless the `LocalIP` prop overrides it, as usual).
3. Check the fix works with a ns6 config backup (this use case should not be affected at all)
4. Check the fix works if NethVoice is installed AFTER restore-config (this is really a corner case!)
5. Check the fix works if NethVoice is installed BEFORE starting the restore-config procedure.

**Test case 3**

After test case 1 and 2, execute

    signal-event post-restore-data

...then check the FreePBX UI is still working, and asterisk & nethcti-server are running.

**Comment by gsanchietti** [comment]
There are still some cases when the restore fails.
The issue needs further investigation.

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethserver-nethvoice14-14.7.1-1.7.g6b50b19.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethserver-nethvoice14-14.7.1-1.7.g6b50b19.ns7.noarch.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethserver-nethvoice14-14.7.1-1.10.ga644074.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethserver-nethvoice14-14.7.1-1.10.ga644074.ns7.noarch.rpm)

**Comment by gsanchietti** [comment]
All test cases seems ok, but I would like to wait a little bit more to gather more feedback.

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethcti-server3-3.20.2-1.5.gbf26a9c.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-3.20.2-1.5.gbf26a9c.ns7.x86_64.rpm) 
- nethcti-server3-debuginfo-3.20.2-1.5.gbf26a9c.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-debuginfo-3.20.2-1.5.gbf26a9c.ns7.x86_64.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethcti-server3-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm) 
- nethcti-server3-debuginfo-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethcti-server3-debuginfo-3.20.2-1.4.gae4bab9.ns7.x86_64.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-testing`:
- nethserver-nethvoice14-14.7.1-1.6.g87f0ac7.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-testing/x86_64/Packages/nethserver-nethvoice14-14.7.1-1.6.g87f0ac7.ns7.noarch.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-updates`:
- nethcti-server3-3.20.3-1.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-updates/x86_64/Packages/nethcti-server3-3.20.3-1.ns7.x86_64.rpm) 
- nethcti-server3-debuginfo-3.20.3-1.ns7.x86_64.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-updates/x86_64/Packages/nethcti-server3-debuginfo-3.20.3-1.ns7.x86_64.rpm)

**Comment by nethbot** [comment]
in `7.8.2003/nethesis-updates`:
- nethserver-nethvoice14-14.7.2-1.ns7.noarch.rpm [x86_64](http://packages.nethesis.it/nethserver/enterprise/7.8.2003/nethesis-updates/x86_64/Packages/nethserver-nethvoice14-14.7.2-1.ns7.noarch.rpm)

**Comment by Stell0** [comment]
Support reported that this issue still exists because freepbxuser mysql password in /etc/freepbx.conf differs from the ones in database

**Comment by DavidePrincipi** [comment]
@Stell0 it seems a related, new bug. Please open a new issue with reproducible steps and proper description, thank you.

Possibly use the nethserver issue tracker if not limited to NethVoice
